tidy3d/components/grid/mesher.py

```diff
@@ -223,12 +228,12 @@ def parse_structures(
             # if there are any enforced structure in the interval, use the last structure
             if max(intervals["structs"][coord_ind]) >= num_unenforced:
                 max_step = structure_steps[max(intervals["structs"][coord_ind])]
-                max_steps.append(float(max_step))
+                max_steps.append(max_step)
             # otherwise, define the max step as the minimum over all medium steps
             # of media in this interval
             else:
                 max_step = np.amin(structure_steps[intervals["structs"][coord_ind]])
-                max_steps.append(float(max_step))
+                max_steps.append(max_step)
```

Comment:
yea I wasn't sure either, but similar to `math.isclose()`, it was breaking autograd and when I removed it everything w worked fine.

And yea the other changes are cosmetic, but actually the `make_shapely_box()` was a remnant from when I was trying to make this autograd compatible. If it doesn't bother you, maybe we can just leave it in case we revisit this?